packages/connect-core/src/entities/App.ts

```diff
@@ -25,92 +31,63 @@ export interface AppData {
+  readonly name?: string
+  readonly registry?: string
+  readonly registryAddress!: string
+  readonly repoAddress?: string
+  readonly version?: string
+  #metadata!: Metadata
 
-  constructor(connector: ConnectorInterface) {
+  constructor(
+    data: AppData,
+    metadata: Metadata,
+    connector: ConnectorInterface
+  ) {
     super(connector)
 
-    this.#created = false
-  }
+    this.address = data.address
+    this.appId = data.appId
+    this.codeAddress = data.codeAddress
+    this.contentUri = data.contentUri || undefined
+    this.isForwarder = data.isForwarder ?? undefined
+    this.isUpgradeable = data.isUpgradeable ?? undefined
+    this.kernelAddress = data.kernelAddress
+    this.name = data.name
+    this.registry = data.registry || undefined
+    this.registryAddress = data.registryAddress
+    this.repoAddress = data.repoAddress
+    this.version = data.version
 
-  get created(): boolean {
-    return this.#created
+    this.#metadata = metadata
   }
 
-  async create({ artifact, manifest, ...data }: AppData): Promise<void> {
-    if (!this.#created) {
-      this.#created = true
-
-      const {
-        appName,
-        path,
-        functions,
-        deprecatedFunctions,
-        abi,
-      }: AragonArtifact = await resolveMetadata(
-        'artifact.json',
-        data.contentUri || undefined,
-        artifact
-      )
-
-      const {
-        author,
-        description,
-        start_url: htmlUrl,
-        icons,
-        source_url: sourceUrl,
-      }: AragonManifest = await resolveMetadata(
-        'manifest.json',
-        data.contentUri || undefined,
-        manifest
-      )
-
-      this.abi = abi
-      this.address = data.address
-      this.appId = data.appId
-      this.appName = appName
-      this.author = author
-      this.codeAddress = data.codeAddress
-      this.contentUri = data.contentUri || undefined
-      this.contractPath = path
-      this.deprecatedIntents = deprecatedFunctions
-      this.description = description
-      this.icons = icons
-      this.intents = functions
-      this.isForwarder = data.isForwarder ?? undefined
-      this.isUpgradeable = data.isUpgradeable ?? undefined
-      this.htmlUrl = htmlUrl
-      this.kernelAddress = data.kernelAddress
-      this.name = data.name
-      this.registry = data.registry || undefined
-      this.registryAddress = data.registryAddress
-      this.repoAddress = data.repoAddress
-      this.version = data.version
-      this.sourceUrl = sourceUrl
-    }
+  static async create(
+    data: AppData,
+    connector: ConnectorInterface
+  ): Promise<App> {
+    const artifact: AragonArtifact = await resolveMetadata(
+      'artifact.json',
+      data.contentUri || undefined,
+      data.artifact
+    )
+
+    const manifest: AragonManifest = await resolveMetadata(
+      'manifest.json',
+      data.contentUri || undefined,
+      data.manifest
+    )
+
+    const metadata: Metadata = [artifact, manifest]
```

Comment:
Now we have a clear distinction between data and metadata. Probably during creation as well we will handle the overrides of the few contract that miss repos on aragonPM (Kernel, ACL, EVM, etc)